src/data/set/lattice.lean

```diff
@@ -80,6 +80,22 @@ notation `⋂` binders `, ` r:(scoped f, Inter f) := r
   -- TODO: more rewrite rules wrt forall / existentials and logical connectives
   -- TODO: also eliminate ∃i, ... ∧ i = t ∧ ...
 
+lemma exists_set_mem_of_union_eq_top {ι : Type*} (t : set ι) (s : ι → set β)
+  (w : (⋃ i ∈ t, s i) = ⊤) (x : β) :
+  ∃ (i ∈ t), x ∈ s i :=
+begin
```

Comment:
Is this worth generalizing to
```suggestion
lemma exists_set_mem_of_union_eq_top {ι : Type*} (t : ι → prop) (s : ι → set β)
  (w : (⋃ i (h : t i), s i) = ⊤) (x : β) :
  ∃ i (h : t i), x ∈ s i :=
begin
```

Reply:
I guess I'm disinclined to, it just seems an even more specific statement (even though it is more general!).